src/Microsoft.AspNet.Mvc.Core/ControllerActionDescriptorBuilder.cs

```diff
@@ -284,6 +290,18 @@ private static ParameterDescriptor CreateParameterDescriptor(ParameterModel para
             return parameterDescriptor;
         }
 
+        private static ParameterDescriptor CreateParameterDescriptor(PropertyModel property)
+        {
+            var parameterDescriptor = new ParameterDescriptor()
+            {
+                BinderMetadata = property.BinderMetadata,
+                Name = property.PropertyName,
+                ParameterType = property.PropertyInfo.PropertyType,
+            };
```

Comment:
AH HA. See, you're ignoring `PropertyModel.Name` tricky. Should we make `PropertyModel.Name` readonly?